sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -195,7 +195,7 @@ object Literal {
     case TimestampNTZType => create(0L, TimestampNTZType)
     case it: DayTimeIntervalType => create(0L, it)
     case it: YearMonthIntervalType => create(0, it)
-    case StringType => Literal("")
+    case st: StringType => Literal(UTF8String.fromString(""), st)
```

Comment:
Why do we need this change?

Reply:
The new and old one are not equal as I can see. The deleted one creates a literal w/ the default collation, but new one uses the specified from the argument.

Reply:
That's right. set intersection tests found an interesting issue that we fail to create a default value in case of StringType with collation.